src/maxflow.rs

```diff
@@ -1 +1,251 @@
+use crate::internal_queue::SimpleQueue;
+use std::cmp::min;
+use std::iter;
+use std::ops::{Add, AddAssign, Sub, SubAssign};
 
+// Maybe it should be in a separate module
+pub trait MfCapacity:
+    Copy + Ord + Add<Output = Self> + AddAssign + Sub<Output = Self> + SubAssign
+{
+    fn zero() -> Self;
+    fn max_value() -> Self;
+}
+
+impl MfCapacity for i32 {
```

Comment:
Maybe use macro for these impls?

Reply:
This is a preliminary implementation.  The `internal_type_traits` implementation (#19) provides with more wide variety of features, so I will replace with it once it is merged.